### ISSUE TITLE
Disabling optionals_for_ungrouped_worlds breaks optionals 

### ISSUE THREAD

**Issue by LloydTao** [Multiverse/Multiverse-Inventories#367]
Hey, I'm trying to set up Multiverse so that I have consistency between the `world`, `world_nether` and `world_the_end` dimensions, while remaining separate from `lobby`.

Toggling `last_location` will keep the player in the correct dimension, even if they try to teleport to a different dimension. However, it becomes impossible to exit the dimension.

Using `/mvm set respawnWorld world`, `/mv modify set respawnWorld world`, `/mv tp world`, `/kill`, or even just jumping into the End fountain, will not work. Toggling off `last_location` and then using the above will put the player in `world` at their spawn.

I assume something with `last_location` messes up the world group's spawn.

**Comment by nicegamer7** [comment]
Have you disabled `optionals_for_ungrouped_worlds`?

**Comment by LloydTao** [comment]
> Have you disabled `optionals_for_ungrouped_worlds`?

Yeah, as well as changing the world group's shares from `all`, to `stats` and `inventory`, with  `optionals_for_ungrouped_worlds` both enabled and disabled. 

**Logs**
Enable last_location.
Step into End portal.
```
2020-05-17 20:03:20 [INFO] [Multiverse-Core-Debug] Player 'TVAsriel' was allowed to go to 'lobby_the_end' because enforceaccess is off.
2020-05-17 20:03:20 [INFO] [Multiverse-Core-Debug] TravelAgent not available for PlayerPortalEvent for TVAsriel
2020-05-17 20:03:20 [INFO] [Multiverse-Core-Debug] Player 'TVAsriel' was allowed to go to 'lobby_the_end' because enforceaccess is off.
2020-05-17 20:03:20 [INFO] [Multiverse-Core-Debug] MV-Core is allowing Player 'TVAsriel' to go to 'lobby_the_end'.
2020-05-17 20:03:20 [INFO] [Multiverse-Core-Debug] MV-Core is allowing Player 'TVAsriel' to go to 'lobby_the_end'.
2020-05-17 20:03:20 [INFO] [Multiverse-Core-Debug] Handling gamemode for player: TVAsriel, Changing to SURVIVAL
```
Use /kill.
Respawn.
```
2020-05-17 20:04:04 [INFO] [Multiverse-Core-Debug] MV-Core is allowing Player 'TVAsriel' to go to 'world'.
2020-05-17 20:04:04 [INFO] [Multiverse-Core-Debug] Handling gamemode for player: TVAsriel, Changing to SURVIVAL
2020-05-17 20:04:04 [INFO] [Multiverse-Core-Debug] Player 'TVAsriel' was allowed to go to 'lobby_the_end' because enforceaccess is off.
2020-05-17 20:04:04 [INFO] [Multiverse-Core-Debug] TravelAgent not available for PlayerPortalEvent for TVAsriel
2020-05-17 20:04:04 [INFO] [Multiverse-Core-Debug] Player 'TVAsriel' was allowed to go to 'lobby_the_end' because enforceaccess is off.
2020-05-17 20:04:04 [INFO] [Multiverse-Core-Debug] MV-Core is allowing Player 'TVAsriel' to go to 'lobby_the_end'.
2020-05-17 20:04:04 [INFO] [Multiverse-Core-Debug] MV-Core is allowing Player 'TVAsriel' to go to 'lobby_the_end'.
2020-05-17 20:04:05 [INFO] [Multiverse-Core-Debug] Handling gamemode for player: TVAsriel, Changing to SURVIVAL
```
Still in The End.
Disable last_location.
Use /kill.
```
2020-05-17 20:05:26 [INFO] [Multiverse-Core-Debug] Handling gamemode for player: TVAsriel, Changing to SURVIVAL
```
Back in World, but not at spawn point ("Your home bed was missing or obstructed").
Use /kill again.
Spawn at my bed in World, as expected.

**Comment by nicegamer7** [comment]
This is known issue we're working on right now, which is that optional sharables do not work as expected when `optionals_for_ungrouped_worlds` is disabled. Sorry for the trouble.

**Comment by LloydTao** [comment]
> This is known issue we're working on right now, which is that optional sharables do not work as expected when `optionals_for_ungrouped_worlds` is disabled. Sorry for the trouble.

It's cool, thank you for the quick responses.

**Comment by ArvickC** [comment]
Is there an estimated time when this will be resolved?

**Comment by nicegamer7** [comment]
Unfortunately, no, there's no estimate. It's a tricky bug to figure out.

**Comment by nicegamer7** [comment]
I have an experimental fix for this issue available in [this](https://github.com/Multiverse/Multiverse-Inventories/pull/394#issuecomment-703764502) test build. But *please* read the warning.

**Comment by nicegamer7** [comment]
This should be fixed in our latest CI build. It'd be nice if someone could confirm.

**Comment by nicegamer7** [comment]
This should be fixed in version 4.2.2.
